### ISSUE TITLE
mips-bsd-user fails to fetch distfiles

### ISSUE THREAD

**Issue by seanbruno** [seanbruno/qemu-bsd-user#28]
MIPS builds are broken at the moment if distfiles haven't already been
fetched.  The issue is that r308996 added AI_ADDRCONFIG to host look ups
making the silent failure in MIPS emulation a critical failure.  e.g.
"ifconfig -a" never has worked, and now its required.  I'm working on
fixing this but there's a lot of flipping/bopping/twisting required to
make the entire data structure big/little endian safe at the moment.

**Comment by dnelson-1901** [comment]
I have a working patch for this.  getifaddrs() uses a routing sysctl to fetch the list of interfaces, so I wrapped  and converted just that one sysctl.  And yes, there was a lot of field-swapping required.  A  list of variable-length routing messages, each with a list of variable-length specially-aligned sockaddr structs.  Feel free to remove all the debugging printfs I used to prove I was actually doing stuff right  :)   As a side-effect, ifconfig mostly works, and netsat -i prints useful info.  The diff is at https://github.com/dnelson-1901/qemu-bsd-user/commit/2171643b1b28bd0849244484f9d434bf664a5664 .
